Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -494,6 +503,21 @@ Eigen::Vector3d ComponentInfo::scaleFactor(const size_t componentIndex) const {
   return (*m_scaleFactors)[componentIndex];
 }
 
+std::string ComponentInfo::name(const size_t componentIndex) const {
+  return (*m_names)[componentIndex];
+}
+
+size_t ComponentInfo::indexOf(const std::string &name) const {
```

Comment:
I feel that this method is not providing a good interface, since it is misleading. It implies a unique mapping from `name` to index exists, but it does not.

A method with this signature should maybe throw if there are multiple components with the same name. Maybe we need a method returning all indices for a given name?

Furthermore, the reverse search makes the behavior different from the old `ComAssembly::getComponentByName`, which return the *first* match, instead of the last.

Reply:
@SimonHeybrock I believe that naming was supposed to be unique but never enforced as such. As the reverse search will be generally faster, I would prefer to keep it and make fixes as and where problems are found. Loosing the ability to terminate the search (after a match has been found) could affect performance quite badly.

How about we instead create a map with component names as part of the `ComponentInfo` creation and throw if any duplicates exist. I did this to find and eliminate all Detector ID duplicates. The alternative would be to store a multimap, but I'm less keen on that.

Reply:
I would avoid introducing complications like maps if possible. For starters, renaming the method to something like `indexOfAny` might be enough?

Reply:
@SimonHeybrock I think that would probably be enough. If we did find that the wrong (first, last whatever) index was being returned, I think my first question would be why are names being duplicated in this particular IDF.